index.js

```diff
@@ -386,6 +391,81 @@ exports.getLastThreadPost = function (thread) {
   return msg
 }
 
+/* post revision utils */
+
+exports.getRevisions = function(ssb, thread, callback) {
+  function collectRevisions(thread) {
+    // this function walks the revisions of a given thread, collecting them up
+    // asyncly
+
+    var threadRevisions = thread.related
+        .map(function(relatedMsg) {
+      
+        if (relatedMsg.value.content.type === 'post-edit' &&
+            // ^ make sure it's an edit
+            relatedMsg.value.content.revision &&
+            // ^ may be unnecessary if schema validates
+            (relatedMsg.value.content.revision === thread.key ||
+             relatedMsg.value.content.root     === thread.key)
```

Comment:
`.relatedMessages()` takes a message ID and gets the full tree of messages that link to it. The concept of a thread is somewhat abstract, but usually we're referring to a "post thread," as in the tree of the discussion. So, in a `type: post`, the `root` link goes to the first post in the discussion.

> So each revision is a "revision thread"

You mean, each post can have its own "revision thread?"